packages/flutter_tools/templates/plugin/.idea/libraries/Dart_SDK.xml.tmpl

```diff
@@ -0,0 +1,25 @@
+<component name="libraryTable">
+  <library name="Dart SDK">
+    <CLASSES>
+      <root url="file://{{dartSdk}}/lib/async" />
+      <root url="file://{{dartSdk}}/lib/collection" />
+      <root url="file://{{dartSdk}}/lib/convert" />
+      <root url="file://{{dartSdk}}/lib/core" />
+      <root url="file://{{dartSdk}}/lib/developer" />
+      <root url="file://{{dartSdk}}/lib/html" />
+      <root url="file://{{dartSdk}}/lib/indexed_db" />
+      <root url="file://{{dartSdk}}/lib/io" />
+      <root url="file://{{dartSdk}}/lib/isolate" />
+      <root url="file://{{dartSdk}}/lib/js" />
+      <root url="file://{{dartSdk}}/lib/js_util" />
+      <root url="file://{{dartSdk}}/lib/math" />
+      <root url="file://{{dartSdk}}/lib/mirrors" />
+      <root url="file://{{dartSdk}}/lib/svg" />
+      <root url="file://{{dartSdk}}/lib/typed_data" />
+      <root url="file://{{dartSdk}}/lib/web_audio" />
+      <root url="file://{{dartSdk}}/lib/web_gl" />
+      <root url="file://{{dartSdk}}/lib/web_sql" />
+    </CLASSES>
```

Comment:
Here's a start:

This is how the Dart plugin writes out the location of the Dart SDK in an IDEA project. It was changed recently - this used to be a global setting and it's now a per-project setting. It lists all core libraries because the same SDK can be used to run both Dart and Flutter code. I don't know how often the Dart plugin updates the list from the Dart SDK itself.

We don't expect users to edit this file. That's why it's in the .idea directory.

It's also used by the Flutter plugin; we derive the location of the Flutter SDK from the Dart SDK.

We've been discussing other ways to determine the location of the Flutter SDK. See discussion in #10096.